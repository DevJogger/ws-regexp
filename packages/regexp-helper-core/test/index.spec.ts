import { toUnicode, surrogatePair, toHex, unicodeUnEscape } from '../index';

describe(`𠮷`, () =>
{
	const c = '𠮷';

	[
		c,
		toUnicode(c),
		toUnicode(c, true),
	].forEach(function (s)
	{
		const r = new RegExp(`^[${s}]$`, 'u');

		test(r.toString(), function ()
		{
			expect(r.test(c)).toBeTruthy();
			expect(r).toMatchSnapshot();
		});
	});

	test(`codePointAt`, () =>
	{

		let actual = c.codePointAt(0);
		let expected = 0x20bb7;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`toUnicode`, () =>
	{

		let actual = toUnicode(c.codePointAt(0));
		let expected = '\\u{20bb7}';

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`surrogatePair`, () =>
	{

		let actual = surrogatePair(c.codePointAt(0));
		let expected = '\\u{20bb7}';

		let c0 = c.charCodeAt(0);
		let c1 = c.charCodeAt(1);

		expect(actual[0]).toStrictEqual(c0);
		expect(actual[1]).toStrictEqual(c1);

		expect(actual.h).toStrictEqual(c0);
		expect(actual.l).toStrictEqual(c1);

		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

		expect(toHex(actual[0])).toStrictEqual('d842');
		expect(toHex(actual[1])).toStrictEqual('dfb7');

	});

})

describe(`unicodeUnEscape`, () =>
{
	test(`\\u{20bb7} => 𠮷`, async function ()
	{
		expect(unicodeUnEscape('\\u{20bb7}')).toStrictEqual('𠮷');
		expect(unicodeUnEscape(toUnicode('𠮷'))).toStrictEqual('𠮷');
		expect('\u{20bb7}').toStrictEqual('𠮷');
	});

	test(`Hello world`, async function ()
	{
		expect(unicodeUnEscape('\\u{48}\\u{65}\\u{6c}\\u{6c}\\u{6f}\\u{20}\\u{77}\\u{6f}\\u{72}\\u{6c}\\u{64}')).toStrictEqual('Hello world');
	});
});
