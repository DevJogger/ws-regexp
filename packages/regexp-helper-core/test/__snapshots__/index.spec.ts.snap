// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`𠮷 /^[\\u{20bb7}]$/u 1`] = `/\\^\\[\\\\u\\{20bb7\\}\\]\\$/u`;

exports[`𠮷 /^[\\ud842\\udfb7]$/u 1`] = `/\\^\\[\\\\ud842\\\\udfb7\\]\\$/u`;

exports[`𠮷 /^[𠮷]$/u 1`] = `/\\^\\[𠮷\\]\\$/u`;

exports[`𠮷 codePointAt 1`] = `134071`;

exports[`𠮷 surrogatePair 1`] = `
[
  55362,
  57271,
]
`;

exports[`𠮷 toUnicode 1`] = `"\\u{20bb7}"`;
